src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -968,7 +974,7 @@ internal virtual HttpClient GetHttpClient(bool handleRedirect)
             }
 
             // This indicates GetResponse will handle redirects.
-            if (handleRedirect)
+            if (handleRedirect || PreserveHTTPMethodOnRedirect)
```

Comment:
I suggest changing the code block as in your another PR to avoid merge conflict.

Reply:
I will change it after we merge the other PR